### ISSUE TITLE
Running shell scripts that wrapped CLIs

### ISSUE THREAD

**Issue by martijnarts** [1Password/shell-plugins#120]
### op CLI version

2.10.0

### Goal or desired behavior

I have a shell script that runs `doctl`, something like:

```bash
#!/usr/bin/env bash

doctl account get
```

### Current behavior

It currently will output:

```
$ ./bin/test.sh
Error: GET https://api.digitalocean.com/v2/account: 401 (request "...") Unable to authenticate you
```

It should just work, though.

**Comment by martijnarts** [comment]
Workaround is replacing the line with `op plugin run doctl account get`

**Comment by mrjones2014** [comment]
I think this is because the `#!/usr/bin/env bash` shebang line causes it to run in a subshell in which `~/.config/op/plugins.sh` would not have been `source`d.

As far as I know there isn't really any way around this other than having your script check if `~/.config/op/plugins.sh` exists, and if so, source it again.

**Comment by mrjones2014** [comment]
Alternatively, you could write it as a shell function in your `~/.bashrc`/`~/.zshrc`/`~/.config/fish/config.fish` instead of having it be a script.

e.g. for bash/zsh

```bash
mydoctlalias() {
  doctl account get
}
```

or for Fish shell:

```fish
function mydoctlalias
  doctl account get
end
```

**Comment by cussiol** [comment]
Those workarounds only work when you can modify the script. As 1Password Shell Plugins are based on `alias`, it seems a significant limitation for actual usage.

It is worth to think an alternative official way to make it work.

One way I thought is something like `op plugins run <plugins> -- ./script`

**Comment by SimonBarendse** [comment]
> I think this is because the #!/usr/bin/env bash shebang line causes it to run in a subshell in which ~/.config/op/plugins.sh would not have been sourced.

This is correct. Furthermore, 1Password Shell Plugins are (currently) build to be used in interactive shells. Besides the problem you're now running into of the aliases not having been sourced, the plugin might also need to prompt (e.g. when you don't have a default credential configured), which is only possible in an interactive terminal.

---

I agree with both these comments:

> It might be worth adding something like this to the dev documentation.

> Those workarounds only work when you can modify the script. As 1Password Shell Plugins are based on alias, it seems a significant limitation for actual usage.
> It is worth to think an alternative official way to make it work.

**Comment by martijnarts** [comment]
I start all my scripts with this now:

```bash
if ! [ -x "$(command -v op)" ]; then
    echo >&2 "Error: 1Password CLI not installed, find it at: https://developer.1password.com/docs/cli/get-started#install"
    exit 1
fi

export OP_ACCOUNT=$(op account list | grep '<SOMETHING>.1password.com' | awk '{print $3}')
```

**Comment by cussiol** [comment]
@SimonBarendse I tried that workaround but got an error:

```sh
$ op plugin run -- /usr/local/bin/aws
[ERROR] 2023/02/09 01:58:06 unknown plugin: /usr/local/bin/aws
$ op plugin run /usr/local/bin/aws
[ERROR] 2023/02/09 01:58:09 unknown plugin: /usr/local/bin/aws
```

**Comment by kjvalencik** [comment]
My workaround for terraform is a wrapper script that calls `aws configure export-credentials --format env` before executing the terraform command.

```sh
#!/bin/bash

eval "$(op plugin run -- aws configure export-credentials --format env)"

/usr/local/bin/terraform "${@}"
```

I have an alias to this script set to `terraform` in my shell config.

**Comment by AndyTitu** [comment]
@lypanov We have just released our new[ beta ](https://app-updates.agilebits.com/product_history/CLI2#v2190001)solution for the [Terraform shell plugin](https://developer.1password.com/docs/cli/shell-plugins/terraform). This allows AWS to work with Terraform out of the box, and also comes up with a general solution for authenticating terraform providers. Would be awesome if you could give it a try and let us know how/if this simplifies your workflows by a bunch.

**Comment by lypanov** [comment]
> @lypanov We have just released our new[ beta ](https://app-updates.agilebits.com/product_history/CLI2#v2190001)solution for the [Terraform shell plugin](https://developer.1password.com/docs/cli/shell-plugins/terraform). This allows AWS to work with Terraform out of the box, and also comes up with a general solution for authenticating terraform providers. Would be awesome if you could give it a try and let us know how/if this simplifies your workflows by a bunch.

Sorry for the delay in response, was recovering from illness. This works really well now thank you for the solution!

**Comment by scottisloud** [comment]
Hi everyone, 

Looks like we have a conclusive answer and way forward here so I'll be closing this issue out. 

Thanks all!
